docs/src/guides/scalar_index.md

```diff
@@ -0,0 +1,109 @@
+# Building Scalar Index
+
+Similar to many SQL databases, LanceDB supports several types of Scalar indices to accelerate search
+over scalar columns.
+
+- `BTREE`: The most common type is BTREE. This index is inspired by the btree data structure
+  although only the first few layers of the btree are cached in memory.
+  It will perform well on columns with a large number of unique values and few rows per value.
+- `BITMAP`: this index stores a bitmap for each unique value in the column.
+  This index is useful for columns with a small number of unique values and many rows per value.
```

Comment:
heh the mathematician in me thinks finite -> could be in trillions.

How about `limited` instead of `finite`, and give a parenthetical "(<5,000)" for added clarity?